### DISCUSSION TITLE
AI driven language servers don't work

### DISCUSSION THREAD

**Discussion by fazibear** [helix-editor/helix#7680] [Troubleshooting] [answered]
### Summary

Tried to connect those language server to helix:

Github Pilot LS: https://github.com/jfcherng/copilot-node-server
Codeium: https://github.com/Exafunction/codeium/releases/

Used nvim to authenticate them, but none of them actually works.

Any clues why? How they are different from analyser for example?

Even wrote a piece of code to watch whole conversation, looks like editor don't ask for completion (even if ctl-x pressed)

### Platform

macOS

### Terminal Emulator

wezterm

### Helix Version

23.05

**Comment by the-mikedavis** [comment]
What does the log file say? https://github.com/helix-editor/helix/wiki/FAQ#access-the-log-file

**Reply by kirawi** [reply]
AFAIK the copilot LSP uses non-standard stuff, hence why this fork was made: https://github.com/helix-editor/helix/pull/6865

**Reply by gabydd** [reply, marked as answer]
Yep, and so does codeium it seems, from there editor plugins it seems like they make a ``GetCompletions`` request which is not part of the language server spec. (we only support things from the official spec). This looks a bit similar to how copilot does it so you could check out if you could get the above linked fork to work with codeium instead of copilot if you want to that. it won't be part of core as discussed in other issues
